[SDK-3821] React beta quickstart (#10188)
auth0/docs: articles/quickstart/spa/react-beta/interactive.md

  callback: 'http://localhost:3000',
  returnTo: 'http://localhost:3000',
  webOriginUrl: 'http://localhost:3000',
  showWebOriginInfo: true
}) %>

## Install the Auth0 React SDK {{{ data-action=code data-code="index.js#13:22" }}}

Auth0 provides a [React SDK](https://github.com/auth0/auth0-react) (auth0-react.js) to simplify the process of implementing Auth0 authentication and authorization in React apps.

Install the Auth0 React SDK by running the following commands in your terminal:

```bash
cd <your-project-directory>
npm install @auth0/auth0-react@beta
```

### Configure the Auth0Provider component

For the SDK to function properly, you must set the following properties in the Auth0Provider component:

- `domain`: The domain of your Auth0 tenant. Generally, you can find this in the Auth0 Dashboard under your Application's Settings in the Domain field. If you are using a [custom domain](https://auth0.com/docs/custom-domains), you should set this to the value of your custom domain instead.
- `clientId`: The ID of the Auth0 Application you set up earlier in this quickstart. You can find this in the Auth0 Dashboard under your Application's Settings in the Client ID field.

> - `authorizationParams.redirectUri`: The URL in your application that you would like Auth0 to redirect users to after they have authenticated. This corresponds to the callback URL you set up earlier in this quickstart. You can also find this value in the Auth0 Dashboard under your Application's Settings in the Callback URLs field. Make sure what you enter in your code matches what you set up earlier or your users will see an error.

Reviewer: ```suggestion
- `authorizationParams.redirect_uri`: The URL in your application that you would like Auth0 to redirect users to after they have authenticated. This corresponds to the callback URL you set up earlier in this quickstart. You can also find this value in the Auth0 Dashboard under your Application's Settings in the Callback URLs field. Make sure what you enter in your code matches what you set up earlier or your users will see an error.
```

:::
::::

## Add Login to Your Application {{{ data-action=code data-code="login.js#4:7" }}}

Now that you have configured your Auth0 Application and the Auth0 React SDK, you need to set up login for your project. To do this, you will use the SDK’s loginWithRedirect() method to create a login button that redirects users to the Auth0 Universal Login page. After a user successfully authenticates, they will be redirected to the callback URL you set up earlier in this quickstart.

<%= include('../../_includes/_auth0-react-classes-info.md') %>

Create a new file in your application called `login.js` for the login button component, and copy in the code from the interactive panel to the right, which contains the logic needed for login. Then, update your `index.js` file to include the new login button.

::::checkpoint

:::checkpoint-default

You should now be able to log in or sign up using a username and password.

Click the login button and verify that:
* your React Application redirects you to the Auth0 Universal Login page
* you can log in or sign up

> * Auth0 redirects you to your application using the value of the `authorizationParams.redirectUri` you used to configure the `Auth0Provider`

Reviewer: ```suggestion
* Auth0 redirects you to your application using the value of the `authorizationParams.redirect_uri` you used to configure the `Auth0Provider`
```

:::

:::checkpoint-failure
Sorry about that. Here's a couple things to double check:

> * you configured the correct `authorizationParams.redirectUri`

Reviewer: ```suggestion
* you configured the correct `authorizationParams.redirect_uri`
```
